### PR TITLE
Update DataSourceModuleTest to use yaml configuration to demonstrate …

### PR DESCRIPTION
**Pull request by jmuia**
…end-to-end DataSource binding

### CODE REVIEW

File: misk/src/test/resources/test_data_source_app-common.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+data_sources:
+  databases:
```
Comment (jmuia) [inline misk/src/test/resources/test_data_source_app-common.yaml]: it isn't great to have this additional level of nesting. I think we'd need to update the `DataSourcesConfig` class to implement the `Map` interface so Jackson populates it properly (?).

**Comment by jmuia** [comment]
I wanted an example of pulling the data sources from configuration files. I didn't update the other configs that override properties to test failure cases, but I can if that makes sense.

File: misk/src/test/resources/test_data_source_app-common.yaml (hunk 1 above)
Comment (mmihic) [reply misk/src/test/resources/test_data_source_app-common.yaml]: yeah totally agreed; can it just implement Map, or do we need a custom Yaml unmarshaller for that type?

Comment (jmuia) [reply misk/src/test/resources/test_data_source_app-common.yaml]: Not 100% sure, I'm going to investigate a bit and will follow up with another PR

**Pull request merged**